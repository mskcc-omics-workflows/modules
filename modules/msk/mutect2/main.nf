process MUTECT2 {
    tag "$meta.id"
    label 'process_high'



    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/gatk4:4.5.0.0--py36hdfd78af_0':
        'biocontainers/gatk4:4.5.0.0--py36hdfd78af_0' }"

    input:
    tuple val(meta), path(case_bam), path(control_bam), path(case_bai), path(control_bai)
    tuple path(bed_file), path(fasta_file), path(fasta_index_file), path(fasta_dict_file)

    output:
    tuple val(meta), path("*.mutect2.vcf"), emit: mutect2_vcf

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def args2 = task.ext.args2 ?: ''
    def case_sample_name = task.ext.prefix ?: "${meta.case_id}"
    def control_sample_name = task.ext.prefix ?: "${meta.control_id}"
    def bed_file = bed_file ? "--intervals ${bed_file}" : ''


    """
    gatk "Mutect2" \\
    -R ${fasta_file} \\
    -I ${case_bam} \\
    -I ${control_bam} \\
    -tumor ${case_sample_name} \\
    -normal ${control_sample_name} \\
    ${args} \\
    --output ${meta.id}.mutect2.vcf

    """

    stub:
    """
    touch ${meta.id}.mutect2.vcf

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        gatk "Mutect2": \$(gatk "Mutect2" --version | sed -e "s/gatk "Mutect2" v//g")
    END_VERSIONS
    """
}
