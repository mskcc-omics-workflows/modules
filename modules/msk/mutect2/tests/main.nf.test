
nextflow_process {

    name "Test Process MUTECT2"
    script "../main.nf"
    process "MUTECT2"

    tag "modules"
    tag "modules_msk"
    tag "mutect2"


    test("mutect2 - test - vcf") {

        when {
            process {
                """
                
                input[0] = [
                    [id: 'test', case_id: 'tumor',control_id:  'normal'],
                    file('https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/homo_sapiens/illumina/bam/test.paired_end.recalibrated.sorted.bam', checkIfExists: true),
                    file('https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/homo_sapiens/illumina/bam/test2.paired_end.recalibrated.sorted.bam', checkIfExists: true),
                    file('https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/homo_sapiens/illumina/bam/test.paired_end.recalibrated.sorted.bam.bai', checkIfExists: true),
                    file('https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/homo_sapiens/illumina/bam/test2.paired_end.recalibrated.sorted.bam.bai', checkIfExists: true)
                ]

                input[1] = [
                    file(params.test_data['sarscov2']['genome']['test_bed'], checkIfExists: true),
                    file('https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/homo_sapiens/genome/chr21/sequence/genome.fasta', checkIfExists: true),
                    file('https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/homo_sapiens/genome/chr21/sequence/genome.fasta.fai', checkIfExists: true),
                    file('https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/homo_sapiens/genome/chr21/sequence/genome.dict', checkIfExists: true)
                ]


                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }


    test("mutect2 - vcf - stub") {

        options "-stub"

        when {
            process {
                """
                
                input[0] = [
                    [id: 'test', case_id: 'tumor',control_id:  'normal'],
                    file('https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/homo_sapiens/illumina/bam/test.paired_end.recalibrated.sorted.bam', checkIfExists: true),
                    file('https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/homo_sapiens/illumina/bam/test2.paired_end.recalibrated.sorted.bam', checkIfExists: true),
                    file('https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/homo_sapiens/illumina/bam/test.paired_end.recalibrated.sorted.bam.bai', checkIfExists: true),
                    file('https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/homo_sapiens/illumina/bam/test2.paired_end.recalibrated.sorted.bam.bai', checkIfExists: true)
                ]

                input[1] = [
                    file(params.test_data['sarscov2']['genome']['test_bed'], checkIfExists: true),
                    file('https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/homo_sapiens/genome/chr21/sequence/genome.fasta', checkIfExists: true),
                    file('https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/homo_sapiens/genome/chr21/sequence/genome.fasta.fai', checkIfExists: true),
                    file('https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/homo_sapiens/genome/chr21/sequence/genome.dict', checkIfExists: true)
                ]
                
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
