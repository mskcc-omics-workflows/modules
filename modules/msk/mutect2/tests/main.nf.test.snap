{
    "mutect2 - vcf - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "case_id": "tumor",
                            "control_id": "normal"
                        },
                        "test.mutect2.vcf:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "mutect2_vcf": [
                    [
                        {
                            "id": "test",
                            "case_id": "tumor",
                            "control_id": "normal"
                        },
                        "test.mutect2.vcf:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-10-29T13:29:18.808923827"
    },
    "mutect2 - test - vcf": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "case_id": "tumor",
                            "control_id": "normal"
                        },
                        "test.mutect2.vcf:md5,657ba36c270271c7946dfbe46ddadb16"
                    ]
                ],
                "mutect2_vcf": [
                    [
                        {
                            "id": "test",
                            "case_id": "tumor",
                            "control_id": "normal"
                        },
                        "test.mutect2.vcf:md5,657ba36c270271c7946dfbe46ddadb16"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-10-29T13:47:42.404427021"
    }
}